File: src/ds-visuals/snippets/min-heap/insert.py
def insert(self, data):
	
	if(self.size == self.CAPACITY):
		print("Error- stack full")
	self.size += 1
	self.heap[self.size] = data
	current = self.size
	while(self.heap[current] < self.heap[self.parent(current)]):
		self.swap(current, self.parent(current));
		current = self.parent(current);
	return;

File: src/ds-visuals/snippets/min-heap/insert.c
9
very_fast =
fast =  0, 1, 2, 6, 9
medium = 3, 4, 5, 7, 8
slow = 
very_slow = 

void insert(Heap* heap, int data){
	int current;
	if(heap->size == CAPACITY)
		printf("Error - Heap full\n");
	heap->size++;
	heap->arr[heap->size] = data;
	current = heap->size; 
	while(heap->arr[current] < heap->arr[parent(current)]){
		swap(heap, current, parent(current));
		current = parent(current);
	}
}